### PR TITLE
Nitweb clean

### PR DESCRIPTION
**Pull request by Morriar**
Cleans some misc part of Nitweb. Notable fact, replace the hover on cards by a dropdown menu with caret.

Can be seen here: http://nitweb.moz-code.org/doc/core::array

For now, the dropdown menu allows the user to swtich between signature and documentation. More contextual options in the future.

This looks quirk with the stars rating (left untouched in the new dropdown system) so still need the mouse over. Since the next PR rewrite this part, I will let this as it is.
We load the stars only when the dropdown is selected so we gain some perfs on the home page.

This PR is based on #2265

### CODE REVIEW

**Comment by privat** [comment]
I like the dropdown caret
Also, maybe keep the title (even in a reduced font) so we still know what is the entity on the card.
The doc should be disabled of something if there is only a synopsys since it is already in the front part of the card.

File: share/nitweb/directives/entity/doc.html [hunk 1]
```diff
@@ -1,8 +1,29 @@
...
+			<div class='dropdown'>
+				<button class='btn btn-link dropdown-toggle' type='button' data-toggle='dropdown'>
+					<span class='glyphicon glyphicon-chevron-down'></span>
+				</button>
+				<ul class='dropdown-menu dropdown-menu-right'>
+					<li ng-class='currentTab == "doc" ? "active" : ""'>
+						<a ng-click='currentTab = "doc"'>Doc</a>
+					</li>
+					<li ng-class='currentTab == "grade" ? "active" : ""'>
+						<a ng-click='currentTab = "grade"'>Grade</a>
```
Comment (privat) [inline share/nitweb/directives/entity/doc.html]: I do not know why, but *grade* make me think of beef. http://www.darcysmeats.ca/2014/09/23/understanding-canadian-beef-grades/

**Comment by privat** [comment]
+1

**Comment by jcbrinfo** [comment]
Menu invisible on Firefox.

**Comment by jcbrinfo** [comment]
I don’t know why, but I don’t see the caret you are talking about.

**Comment by Morriar** [comment]
On firefox:

image

Maybe clean your cache? What version are you using?

**Comment by jcbrinfo** [comment]
I even tried with a clean version of Chrome and nothing appeared. Now I see them. Weird. UQAM network I gess.

File: share/nitweb/stylesheets/nitweb.css [hunk 2]
```diff
@@ -29,8 +29,6 @@ a {
 	border: 1px solid #1E9431;
 }
 
-.card, .card-body { overflow: hidden; }
-
```
Comment (jcbrinfo) [inline share/nitweb/stylesheets/nitweb.css]: :heavy_check_mark: in [#2177](https://github.com/nitlang/nit/issues/2177#issuecomment-227482006)

**Comment by jcbrinfo** [comment]
:+1:

**Comment by Bl4ckb0ne** [comment]
not responsive
Responsive/20

**Comment by jcbrinfo** [comment]
@Bl4ckb0ne: We have #2177 for that.

File: src/web/web_mongo.nit [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	#
+	# * key: `mongo.db`
+	# * default: `nitweb`
+	var mongo_db: String is lazy do return value_or_default("mongo.db", "nitweb")
+
+	# Mongo instance
+	var mongo: MongoClient is lazy do return new MongoClient(mongo_uri)
+
+	# Database instance
+	var db: MongoDb is lazy do return mongo.database(mongo_db)
```
Comment (privat) [inline src/web/web_mongo.nit]: ../src/web/web_mongo.nit:42,6--7: Redef Error: `NitwebConfig::db` is an inherited property. To redefine it, add the `redef` keyword.../src/web/web_mongo.nit:42,6--7: Redef Error: `NitwebConfig::db` is an inherited property. To redefine it, add the `redef` keyword.

cf http://gresil.org/jenkins/job/nit_tests/engine=nitg-s,label=deploy_linux/1252/testReport/tests.nitcs.src/nitweb/nitweb/

Comment (Morriar) [inline src/web/web_mongo.nit]: Oups, ça conflicte avec la pr qui remonte ça dans pop_repos

**Comment by privat** [comment]
I will merge the other PR, you can rebase this one after.

**Comment by Morriar** [comment]
test this please

**Comment by privat** [comment]
http://gresil.org/jenkins/view/Int%C3%A9gration/job/nit_tests/engine=nitg-s,label=deploy_linux/1256/testReport/tests.nitcs.src/nitweb/nitweb/

**Pull request merged**